php-nginx/build-scripts/build_php71.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A shell script for installing PHP 7.0.
+set -xe
+
+PHP_SRC=/usr/src/php71
+
+curl -SL "https://downloads.php.net/~ab/php-$PHP71_VERSION.tar.gz" -o php71.tar.gz
+# curl -SL "http://us2.php.net/get/php-$PHP71_VERSION.tar.gz.asc/from/this/mirror" -o php71.tar.gz.asc
+
+##
+# Skipping pgp checks, as there was no detailed view on who signs 7.1 code
+##
+
+# Create combined binary keys
+# cat /gpgkeys/php70/* | gpg --dearmor > /gpgkeys/php70.gpg
+
+# Verify only with specific public keys
+# gpg --no-default-keyring --keyring /gpgkeys/php70.gpg --verify php7.tar.gz.asc
+
+mkdir -p ${PHP_SRC}
+tar -zxf php71.tar.gz -C ${PHP_SRC} --strip-components=1
+rm php71.tar.gz
+# rm php71.tar.gz.asc
+
+# TODO: Install more 3rd party extensions.
+
+# TODO: Use stable version of memcached from pecl once available
+git clone -b php7 https://github.com/php-memcached-dev/php-memcached ${PHP_SRC}/ext/memcached
+
+# No need for jsonc replacement.
+
+mkdir -p ${PHP_SRC}/ext/mailparse
+curl -SL "https://pecl.php.net/get/mailparse" -o mailparse.tar.gz
+tar -zxf mailparse.tar.gz -C ${PHP_SRC}/ext/mailparse --strip-components=1
+rm mailparse.tar.gz
+
+# APCu
+mkdir -p ${PHP_SRC}/ext/apcu
+curl -SL "https://pecl.php.net/get/apcu-5.1.4.tgz" -o apcu.tar.gz
+tar -zxf apcu.tar.gz -C ${PHP_SRC}/ext/apcu --strip-components=1
+rm apcu.tar.gz
+
+# APC compatibility layer for APCu
+mkdir -p ${PHP_SRC}/ext/apcu-bc
+curl -SL "https://pecl.php.net/get/apcu_bc-1.0.3.tgz" -o apcu-bc.tar.gz
+tar -zxf apcu-bc.tar.gz -C ${PHP_SRC}/ext/apcu-bc --strip-components=1
+rm apcu-bc.tar.gz
+
+pushd ${PHP_SRC}
+# Consider adding an equivalent patch in later stages
+# patch -p1 < /build-scripts/php7-parse_str_harden.patch
```

Comment:
Yep, did so yesterday - unfortunately the sources have diverged. - I guess I'll have to replicate it, but my C is .. a little bit rusty, so I postponed it :)